core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -232,6 +232,10 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
 
     def nextQuery(state: EventsBySlicesState): (EventsBySlicesState, Option[Source[EventEnvelope, NotUsed]]) = {
       val newIdleCount = if (state.rowCount == 0) state.idleCount + 1 else 0
+
+      // Note: Found this hard to decipher - start backtracking if we saw no new events in 5 polls and time between
+      // last seen timestamp and last backtracking timestamp is larger than half backtracking window (???)
+      // what is last backtracking timestamp?
```

Comment:
yup, and I'm not sure I have found the right decision criteria yet.

Currently, there are essentially two non-overlapping scans.
1. The normal that search from latest offset until now - 100ms.
2. The backtracking that search from latest backtracking offset until latest normal offset and now - 3sec.

Switching to backtracking if 5 idle (empty) results from previous queries, or if latest backtracking offset is too far bind latest normal offset (halfBacktrackingWindow).